t/31-sshSerial.t

```diff
@@ -0,0 +1,204 @@
+
+my $eagain = [
+    LIBSSH2_ERROR_EAGAIN,
+    'LIBSSH2_ERROR_EAGAIN',
+    'Operation would block'
+];
+
+my $mock_backend = Test::MockObject->new();
+my $mock_ssh     = Test::MockObject->new();
+my $mock_channel = Test::MockObject->new();
+
+$mock_ssh->{error} = undef;
+
+$mock_channel->{write_limits} = [];
+$mock_channel->{write_buffer} = '';
+$mock_channel->{read_queue}   = [];
+$mock_channel->{blocking}     = 1;
+
+$mock_channel->mock(pty => sub {
+        return 1;
+});
+
+$mock_channel->mock(ext_data => sub {
+        return 1;
+});
+
+$mock_channel->mock(shell => sub {
+        return 1;
+});
+
+$mock_channel->mock(blocking => sub {
+        my $self = shift;
+        my $arg  = shift;
+
+        $self->{blocking} = $arg if defined($arg);
+        return $self->{blocking};
+});
+
+$mock_channel->mock(read => sub {
+        my ($self, undef, $size) = @_;
+
+        my $data = shift @{$self->{read_queue}};
+
+        if (!defined($data)) {
+            $mock_ssh->{error} = $eagain unless defined($mock_ssh->{error});
+            return undef;
+        }
+
+        if (length($data) > $size) {
+            unshift @{$self->{read_queue}}, substr($data, $size);
+            $data = substr($data, 0, $size);
+        }
+
+        $_[1] = $data;
+        return length($data);
+});
+
+$mock_channel->mock(write => sub {
+        my ($self, $data) = @_;
+        my $limit = shift @{$self->{write_limits}};
+
+        if (defined($limit) && $limit < 0) {
+            $mock_ssh->{error} = $eagain unless defined($mock_ssh->{error});
+            return undef;
+        }
+
+        $data = substr($data, 0, $limit) if defined($limit);
+        $self->{write_buffer} .= $data;
+        return length($data);
+});
+
+$mock_channel->mock(send_eof => sub {
+        return 1;
+});
+
+$mock_ssh->mock(channel => sub {
+        return $mock_channel;
+});
+
+$mock_ssh->mock(blocking => sub {
+        my $self = shift;
+        my $arg  = shift;
+
+        return $mock_channel->blocking($arg);
+});
+
+$mock_ssh->mock(error => sub {
+        my $self = shift;
+
+        return undef unless defined($self->{error});
+        return ${$self->{error}}[0] if ((caller(0))[5]);
+        return @{$self->{error}};
+});
+
+$mock_ssh->mock(die_with_error => sub {
+        my ($self, $arg);
+
+        die $arg;
+});
+
+$mock_backend->mock(new_ssh_connection => sub {
+        return $mock_ssh;
+});
+
+subtest 'Read test' => sub {
+    $mock_ssh->{error}            = undef;
+    $mock_channel->{write_limits} = [];
+    $mock_channel->{write_buffer} = '';
+    $mock_channel->{read_queue}   = [
+        'First line',
+        'Second line',
+        undef,
+        'Third line'
+    ];
+    $mock_channel->{blocking} = 1;
+
+    my $console = consoles::sshSerial->new(undef, {hostname => 'localhost'});
+    $console->backend($mock_backend);
+    $console->activate();
+    my $screen = $console->screen();
+    my $data;
+    my $ret;
+
+    ok(!$mock_channel->{blocking});
```

Comment:
please add messages as last argument in all Test::More calls, e.g.

```suggestion
    ok !$mock_channel->{blocking}, 'non-blocking by default';
```

Reply:
Fixed. But with a different message.